packages/tokamak/sdk/tasks/deposit-withdraw-ton.ts

```diff
@@ -0,0 +1,281 @@
+import { task, types } from 'hardhat/config'
+import { HardhatRuntimeEnvironment } from 'hardhat/types'
+import '@nomiclabs/hardhat-ethers'
+import 'hardhat-deploy'
+import { predeploys } from '@eth-optimism/core-utils'
+import { BytesLike, ethers } from 'ethers'
+
+import {
+  CrossChainMessenger,
+  MessageStatus,
+  TONBridgeAdapter,
+  NumberLike,
+} from '../src'
+
+console.log('Setup task...')
+
+const privateKey = process.env.PRIVATE_KEY as BytesLike
+
+const l1Provider = new ethers.providers.StaticJsonRpcProvider(
+  process.env.L1_URL
+)
+const l2Provider = new ethers.providers.StaticJsonRpcProvider(
+  process.env.L2_URL
+)
+const l1Wallet = new ethers.Wallet(privateKey, l1Provider)
+const l2Wallet = new ethers.Wallet(privateKey, l2Provider)
+
+const erc20ABI = [
+  {
+    inputs: [
+      { internalType: 'address', name: '_spender', type: 'address' },
+      { internalType: 'uint256', name: '_value', type: 'uint256' },
+    ],
+    name: 'approve',
+    outputs: [{ internalType: 'bool', name: '', type: 'bool' }],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+  {
+    constant: true,
+    inputs: [{ name: '_owner', type: 'address' }],
+    name: 'balanceOf',
+    outputs: [{ name: 'balance', type: 'uint256' }],
+    type: 'function',
+  },
+  {
+    inputs: [{ internalType: 'uint256', name: 'amount', type: 'uint256' }],
+    name: 'faucet',
+    outputs: [],
+    stateMutability: 'nonpayable',
+    type: 'function',
+  },
+]
+
+const ETH = '0xDeadDeAddeAddEAddeadDEaDDEAdDeaDDeAD0000'
+
+const zeroAddr = '0x'.padEnd(42, '0')
+
+let TON = process.env.TON || ''
+let addressManager = process.env.ADDRESS_MANAGER || ''
+let l1CrossDomainMessenger = process.env.L1_CROSS_DOMAIN_MESSENGER || ''
+let l1StandardBridge = process.env.L1_STANDARD_BRIDGE || ''
+let optimismPortal = process.env.OPTIMISM_PORTAL || ''
+let l2OutputOracle = process.env.L1_OUTPUT_ORACLE || ''
```

Comment:
I think it should be changed to L2_OUTPUT_ORACLE.

Reply:
I updated!